torch/_inductor/lowering.py

```diff
@@ -2950,8 +2955,7 @@ def index_put_impl_(self, indices, values, accumulate, check):
     x_size = self.get_size()
     x_ndim = len(x_size)
 
-    # fallback to aten.index_put_, as tl.atomic_add does NOT support int64 or bool
-    if self.get_dtype() in {torch.int64, torch.bool}:
+    if needs_fallback_due_to_atomic_add_limitations(self.get_dtype()):
```

Comment:
`self` can be float, see test failures:

FAILED [0.0248s] inductor/test_torchinductor.py::CpuTests::test_scatter3_cpu - torch._dynamo.exc.BackendCompilerFailed: backend='compile_fx_wrapper' raised:
LoweringException: AttributeError: 'float' object has no attribute 'get_dtype'
  target: aten.scatter.value_reduce

Reply:
if self can be float, how did this code even work before? weren't we previously also unconditionally calling sef.get_dtype()?

Reply:
Sorry, it may be the other callsite. Anyway check the failing tests

Reply:
yep, already fixed that one, i thought there was something i missed here